static/src/javascripts-legacy/projects/common/modules/email/email-article.js

```diff
@@ -205,16 +210,41 @@ define([
 
                 storage.session.set('email-sign-up-seen', 'true');
             }
-        };
+        }
 
     return {
         init: function () {
             if (emailRunChecks.allEmailCanRun()) {
                 // First we need to check the user's email subscriptions
                 // so we don't insert the sign-up if they've already subscribed
                 emailRunChecks.getUserEmailSubscriptions().then(function () {
-                    // Get the first list that is allowed on this page
-                    addListToPage(find(listConfigs, emailRunChecks.listCanRun));
+                    if (ab.isParticipating({id: 'TailorRecommendedEmail'}) &&
+                        ab.isInVariant('TailorRecommendedEmail', 'tailor-recommended')) {
+                        var bwidCookie = cookies.get('bwid') || false;
+
+                        if (bwidCookie) {
+                            var cachedTailorResponse = cookies.get('GU_TAILOR_EMAIL') || false;
+                            if(!cachedTailorResponse) {
+                                tailor.getEmail(bwidCookie).then(function (tailorRes) {
+                                    addListToPage(find(listConfigs, emailRunChecks.listCanRun), 'tailor-recommend:signup');
+                                    mediator.emit('tailor-recommended:insert');
+                                    cookies.add('GU_TAILOR_EMAIL', tailorRes.email, 1)
```

Comment:
Also, there are a few strings (e.g. GU_TAILOR_EMAIL) that might be safer as constants)